python/ray/data/tests/test_dataset.py

```diff
@@ -624,6 +623,23 @@ def test_tensor_array_array_protocol(ray_start_regular_shared):
     )
 
 
+def test_tensor_array_dataframe_repr(ray_start_regular_shared):
```

Comment:
This really is not specific to dataset and a test_pandas.py under ray/air/util/tensor_extensions will likely make more sense (and a util without dedicated testing seems scary).

Reply:
Agreed, but there are a bunch of other tests in this module targeting the tensor extension, and we should have a dedicated PR that moves all of the tests over.

Can we do that in a follow-up PR? I don't want to block a PR that needs to land by the branch cut on a test code move that is lower priority and could happen afterwards.

Reply:
We can start with this single test and take retrospective work as followup.

Reply:
Having test coverage split across multiple modules would be much worse IMO. I'll just move all of the tests in this PR so we can unblock this.